### PR TITLE
W to g guide

### PR DESCRIPTION
**Pull request by sjsrey**
This is a start towards addressing #712.
Seeking feedback and discussion.

It will add a note to the index page:

image

This also adds a dynamically generated rst file that will report on the members for W and Graph, which is the target of the note.

image

In addition, I will add a notebook that builds on the suggestions in #712 to have more of a user facing orientation.

This is intended to be dynamic and assist us as we fully build out the `Graph`  moving forward.

Once we are done with the transition, we can remove this dynamically generated rst file, if so desired.

### CODE REVIEW

**Comment by knaaptime** [comment]
sweet

File: docs/Makefile [hunk 1]
```diff
@@ -12,23 +12,18 @@ BUILDDIR      = _build
...
+
+# Custom target to generate rst files
+generate-rst:
+	@echo "Generating custom rst files..."
+	@python ../scripts/w_g_migration.py
 
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: Makefile
-	@rsync -r --exclude '.ipynb_checkpoints/' ../notebooks/ ./notebooks/
```
Comment (jGaboardi) [inline docs/Makefile:20]: Are these deletions simply relics from our older infrastructure?

Comment (sjsrey) [reply docs/Makefile:20]: Yes

**Review by jGaboardi — APPROVED** [review]

**Comment by martinfleis** [comment]
I like that automatic comparison but in some cases, it may be worth adding a note explaining _why_ there's no match and what is recommended to do instead. E.g Graph does not have `islands` but has `isolates` instead etc.

I also suppose that merge shall wait until the notebook is complete, right?

**Comment by sjsrey** [comment]
> I like that automatic comparison but in some cases, it may be worth adding a note explaining _why_ there's no match and what is recommended to do instead. E.g Graph does not have `islands` but has `isolates` instead etc.

Yes, this was intended to identify all the cases we need to surface in the recommendations. I wasn't sure if we should put the recommendations in these tables, or treat them in the migration notebook where we can have a bit more narrative for the user.


> 
> I also suppose that merge shall wait until the notebook is complete, right?

Definitely.

**Comment by martinfleis** [comment]
>  treat them in the migration notebook where we can have a bit more narrative for the user.

That may be better. We just need to make a link to the notebook clear and visible from the RST with the tables.

**Review by knaaptime — APPROVED** [review]

**Comment by sjsrey** [comment]
This is still WIP. I have split the migration material and member comparisons apart for different audiences (users,devs).
This also reorganizes the notebooks and changes the tutorial to a user-guide.

The idea is to keep the component comparison around to automatically generate the W-G tables. These will get updated each time the docs are built.

Meanwhile, the user migration guide is a notebook that can now be fleshed out for the current state of the Graph.
`docs/user-guide/graph/w_g_migration.ipynb` which can now be updated.

**Pull request merged**